metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableUtil.java

```diff
@@ -0,0 +1,196 @@
+        node.set(DataMetrics.rowCount.getMetricName(), valueNode);
+
+        valueNode = JsonNodeFactory.instance.objectNode();
+        valueNode.put(DataMetricConstants.DATA_METADATA_VALUE, metrics.fileCount());
+        node.set(DataMetrics.fileCount.getMetricName(), valueNode);
+        return node;
+    }
+
+    /**
+     * Implemented BaseMetastoreTables to interact with iceberg library.
+     * Load an iceberg table from a location.
+     */
+    private final class IcebergMetastoreTables extends BaseMetastoreTables {
+        private final String tableLocation;
+
+        IcebergMetastoreTables(final String tableMetadataLocation) {
+            super(conf);
+            this.tableLocation = tableMetadataLocation;
+        }
+
+        @Override
+        public Table create(final Schema schema, final PartitionSpec spec, final String database, final String table) {
+            throw new MetacatNotSupportedException("not supported");
+        }
+
+        @Override
+        public Table create(final Schema schema,
+                            final PartitionSpec spec,
+                            final Map<String, String> properties,
+                            final String table) {
+            throw new MetacatNotSupportedException("Not supported");
+        }
+
+        @Override
+        public Table create(final Schema schema, final PartitionSpec spec, final String tables) {
+            throw new MetacatNotSupportedException("Not supported");
+        }
+
+        @Override
+        public Table load(final String tableName) {
+            final QualifiedName table = QualifiedName.fromString(tableName);
+            return super.load(table.getDatabaseName(), table.getTableName());
+        }
+
+        @Override
+        public BaseMetastoreTableOperations newTableOps(final Configuration config,
+                                                        final String database,
+                                                        final String table) {
+            return newIcebergTableOps(config);
+        }
+
+        /**
+         * Creates a new instance of IcebergTableOps for the given table location.
+         *
+         * @param config a Configuration
+         * @return a new MetacatServerOps for the table
+         */
+        private IcebergTableOps newIcebergTableOps(final Configuration config) {
+            return new IcebergTableOps(tableLocation);
+        }
+
+    }
+
+    /**
+     * Implemented the BaseMetastoreTableOperations to interact with iceberg library.
+     * Read only operations.
+     */
+    private final class IcebergTableOps extends BaseMetastoreTableOperations {
+        private final String location;
+
+        IcebergTableOps(final String location) {
+            super(conf);
+            this.location = location;
+            refresh();
+        }
+
+        @Override
+        public TableMetadata refresh() {
+            refreshFromMetadataLocation(this.location);
```

Comment:
What does this method do?

Reply:
This is called from iceberg runtime to load the iceberg table.